### ISSUE TITLE
ENH: Add optimization method "Barycenter method" to optimizers

### ISSUE THREAD

**Issue by brunolnetto** [scipy/scipy#19567]
### Is your feature request related to a problem? Please describe.

This feature is not related to a problem, but an extension to current library. 

### Describe the solution you'd like.

The here-required optimization method is called the "Barycenter Method". His name originates from calculating the barycenter respective to some Oracle map based on evaluation points. Each point evaluates to a scalar quantity, and the barycenter weight is the exponential of the scaled negative of this value. The higher the value, the lower the weight. It has 
a gradient behaviour, but it derivative-free. I would be glad to have this method incorporated to scipy, since I worked with on my Master thesis. 

[1] https://arxiv.org/abs/2102.10467 

### Describe alternatives you've considered.

Currently, I have this python version: https://github.com/asmove/pybary . I would be very grateful for any feedback in this regard.

 Python

**Comment by j-bowhay** [comment]
Thanks @brunolnetto I would suggest posting this to the mailing list too before starting any work so as not to waste effort. Are you able to provide any benchmarks to show if this is able to outperform the existing methods in scipy on example problems? The method does not seem to be very widespread or highly cited so we would need some motivating examples I suspect

**Comment by brunolnetto** [comment]
Thanks to the fast response.

How do I add it to mailing? :-S

Some methods to compare are Nelder-mead, gradient-descent, Newton-Raphson, stochastic gradient descent. Every method has its hyperparameters, I can stablish some metrics like convergence speed and computation effort, but it would require comparable hyperparameter equivalence. I will see what I do.

**Comment by lucascolley** [comment]
> How do I add it to mailing? :-S

You can start a new thread here https://mail.python.org/archives/list/scipy-dev@python.org/ :)

You can find our guidance for suggesting and implementing new features at [Contributing new code](https://docs.scipy.org/doc/scipy/dev/hacking.html#contributing-new-code).

**Comment by j-bowhay** [comment]
We have a fairly extensive existing set of benchmarks
https://github.com/scipy/scipy/blob/main/benchmarks/benchmarks/optimize.py#L265

**Comment by brunolnetto** [comment]
I tried something like `from scipy.benchmarks.benchmarks.optimize import bench_run`, and got an error. I am not very familiar with the scipy pipelines and workflows. Would you mind writing down the proper code to run the benchmarks? Thanks!

**Comment by mattip** [comment]
Maybe this guide to benchmarks will help https://docs.scipy.org/doc/scipy/dev/contributor/benchmarking.html

**Comment by brunolnetto** [comment]
@mattip It may be some miss-setup of my machine: the very first command run `python dev.py bench` stumbled on below error. I did not proceed further.

```
  File "dev.py", line 210
    console.print("[red bold] msg")
                ^
SyntaxError: invalid syntax
```

**Comment by mattip** [comment]
This may be due to some problem earlier in `dev.py`. Perhaps you have some inadvertent modifications of your sources: make sure `git status` does not show anything unintentional. If you can't solve it, please open another issue with a full reproducer: what version of python, what platform, how you installed the dependencies.

**Comment by lucascolley** [comment]
I'm happy to help debug in a new issue if you would like to avoid creating noise here.
